Makefile

```diff
@@ -14,7 +14,15 @@
 
 MAKEFILE_PATH := $(abspath $(lastword $(MAKEFILE_LIST)))
 MAKEFILE_DIR  := $(dir $(MAKEFILE_PATH))
-NPROC ?= $(shell (nproc))
+
+# Process count with OS detection
+ifeq ($(shell (uname -s)), Linux)
+	NPROC ?= $(shell (nproc))
+endif
+ifeq ($(shell (uname -s)), Darwin)
+	NPROC ?= $(shell (sysctl -n hw.physicalcpu))
```

Comment:
Logicialcpu

Reply:
According to https://developer.apple.com/documentation/foundation/processinfo/1415622-processorcount, fix it with hw.ncpu